src/symbols/Node.tsx

```diff
@@ -34,7 +35,16 @@ export const Node: FC<NodeProps> = ({
   labelFontUrl,
   contextMenu,
   onClick,
-  onContextMenu
+  onContextMenu,
+  renderNode = ({ theme, node, nodeSize, active, opacity, animated }) => (
```

Comment:
We need to look at the performance implications of this callback like this. In my other projects, I give users the ability to pass JSX elements but there is some scope and performance concerns with this. One example is I don't think the r3f hook will work if I were to make my own sphere and then pass it down like this since the scope would be outside.

I kinda feel like it would be better to define some sort of prebuilt interface. For instance something like this:

```
nodeView: 'default' | 'iconSphere';
nodeLabelPosition: 'inside' | 'outside';
```

Reply:
I'm not sure performance has changed - though there could be some `useMemo` trick to memoise the node.

I've got an example in the storybook which swaps the node for a custom element (a torus geometry) and it all seems fine